### PR TITLE
Add Ultimate College of Winterhold - an ICOW and OCOW Patch

### PR DESCRIPTION
**Pull request by themayor897**
Try number 4 at getting this correct....

I have edited the entries for Immersive College of Winterhold and Obscure's College of Winterhold to reflect the fact that there is now a compatibility patch for the two of them. I would greatly appreciate it if someone could look over these changes to verify that nothing else is needed, this is a big patch and I'm sure I'll be making more edits to this as I develop patches. Thanks so much!

Also added patch note for AI Overhaul, updated clean status for ICOW.

### CODE REVIEW

**Comment by themayor897** [comment]
Okay I have no idea what's going on, would someone mind taking a look at this? Thank you!

File: masterlist.yaml [hunk 1]
```diff
@@ -9021,6 +9021,11 @@ plugins:
           - 'Open Cities Skyrim'
           - '[AI Overhaul - Open Cites Compatibility](https://www.nexusmods.com/skyrimspecialedition/mods/38031)'
         condition: 'active("Open Cities Skyrim.esp") and not active("AI Overhaul - Open Cities.esp")'
+      - <<: *patch3rdParty
+        subs:
+          - 'Immersive College of Winterhold'
+          - '[Immersive College of Winterhold - AI Overhaul Patch](https://www.nexusmods.com/skyrimspecialedition/mods/41694)'
+        condition: 'active("CollegeOfWinterholdImmersive.esp") and not active ("AI Overhaul - CWI Patch.esp")'
```
Comment (sibir-ine) [inline masterlist.yaml:9028]:
```suggestion
        condition: 'active("CollegeOfWinterholdImmersive.esp") and not active("AI Overhaul - CWI Patch.esp")'
```

**Comment by themayor897** [comment]
Wohoo all fixed!

File: masterlist.yaml [hunk 2]
```diff
@@ -9021,6 +9021,11 @@ plugins:
           - 'Open Cities Skyrim'
           - '[AI Overhaul - Open Cites Compatibility](https://www.nexusmods.com/skyrimspecialedition/mods/38031)'
         condition: 'active("Open Cities Skyrim.esp") and not active("AI Overhaul - Open Cities.esp")'
+      - <<: *patch3rdParty
+        subs:
+          - 'Immersive College of Winterhold'
+          - '[Immersive College of Winterhold - AI Overhaul Patch](https://www.nexusmods.com/skyrimspecialedition/mods/41694)'
+        condition: 'active("CollegeOfWinterholdImmersive.esp") and not active("AI Overhaul - CWI Patch.esp")'
```
Comment (MacSplody) [inline masterlist.yaml]: I would put this and the change to `CollegeOfWinterholdImmersive.esp`cleaning info in a separate pull request.
```suggestion
```

File: masterlist.yaml [hunk 3]
```diff
@@ -13535,11 +13545,9 @@ plugins:
...
       - Stats
       - Text
-    dirty:
-      - <<: *quickClean
-        crc: 0x80E5FDAC
-        util: '[SSEEdit v4.0.3](https://www.nexusmods.com/skyrimspecialedition/mods/164)'
-        itm: 1
+    clean:
+      - crc: 0x01E10725
+        util: 'SSEEdit v4.0.3'
```
Comment (MacSplody) [inline masterlist.yaml]: Same as above, it would be best to put it in a separate pull request for updating Immersive College together with the new patch message.
```suggestion
    dirty:
      - <<: *quickClean
        crc: 0x80E5FDAC
        util: '[SSEEdit v4.0.3](https://www.nexusmods.com/skyrimspecialedition/mods/164)'
        itm: 1
```

File: masterlist.yaml [hunk 4]
```diff
@@ -13511,6 +13516,11 @@ plugins:
       - <<: *patchIncluded
         subs: [ 'Skyrim Project Optimization Full ESM' ]
         condition: 'active("Skyrim Project Optimization - Full Version.esm") and not active("CWISPOPatchSE.esp")'
+      - <<: *patch3rdParty
+        subs:
+          - 'Obscure''s College of Winterhold'
+          - '[Ultimate College of Winterhold](https://www.nexusmods.com/skyrimspecialedition/mods/44797)'
+        condition: 'active("OCW_Obscure''s_CollegeofWinterhold.esp") and not active("UltimateCollege.esp")'
```
Comment (MacSplody) [inline masterlist.yaml]: This is fine since it is relevant to the Ultimate College patch.
Edit: Suggest removing as it's duplicated in OCW metadata.
```suggestion
```

File: masterlist.yaml [hunk 5]
```diff
@@ -14838,14 +14846,15 @@ plugins:
         condition: 'version("OCW_Obscure''s_CollegeofWinterhold.esp", "1.5.00", <)'
       - name: 'WACCF_Armor and Clothing Extension.esp'
         condition: 'version("OCW_Obscure''s_CollegeofWinterhold.esp", "1.5.00", <)'
+      - name: 'CollegeOfWinterholdImmersive.esp'
+        condition: 'version("OCW_Obscure''s_CollegeofWinterhold.esp", "1.5.00", <)'
```
Comment (MacSplody) [inline masterlist.yaml]: Should probably only apply the after if this patch is being used.
Someone else could release another patch with a different order.
The version might not be needed as well, the `<`means it would only work for version prior to `1.5.00`.
```suggestion
        condition: 'active("UltimateCollege.esp")'
```

File: masterlist.yaml [hunk 6]
```diff
@@ -14838,14 +14846,15 @@ plugins:
...
     inc:
       - 'accessible saarthal and labyrinthian.esp'
       - 'ApepiBetterCollege.esp'
       - 'Better College Students Room.esp'
       - 'College Guards SE JMR.esp'
       - 'CollegeDormDoors.esp'
       - 'CollegeGrounds.esp'
-      - 'CollegeOfWinterholdImmersive.esp'
       - 'CollegeofWinterholdUpgraded.esp'
```
Comment (MacSplody) [inline masterlist.yaml]: We could still keep the inc I think, just turn it off if the patch is being used.
```suggestion
      - name: 'CollegeOfWinterholdImmersive.esp'
        condition: 'not active("UltimateCollege.esp")'
      - 'CollegeofWinterholdUpgraded.esp'
```

File: masterlist.yaml [hunk 7]
```diff
@@ -15082,6 +15091,11 @@ plugins:
       - <<: *patchIncluded
         subs: [ 'Winterhold Restored & WR JKs Skyrim Patch' ]
         condition: 'active("Winterhold Restored.esp") and active("Winterhold Restored - JKs Skyrim Patch.esp") and not active("OCW_WR2_JK_FEPatch.esp")'
+      - <<: *patch3rdParty
+        subs: 
+          - 'Immersive College of Winterhold'
+          - '[Ultimate College of Winterhold](https://www.nexusmods.com/skyrimspecialedition/mods/44797)'
+        condition: 'active("CollegeOfWinterholdImmersive.esp") and not active("UltimateCollege.esp")'
```
Comment (MacSplody) [inline masterlist.yaml]: Once is probably enough to show the patch message, I'd say leave this once and remove the other.

File: masterlist.yaml [hunk 8]
```diff
@@ -18091,6 +18108,15 @@ plugins:
     after:
       - 'Qw_ACE_CRF Patch.esp'
       - 'Qw_WACCF_CRF Patch.esp'
+          
+  - name: 'UltimateCollege.esp'
+    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/44797' ]
+    msg:
+      - <<: *compatNotes
+        subs: [https://www.nexusmods.com/skyrimspecialedition/articles/2761]
```
Comment (MacSplody) [inline masterlist.yaml]: [Installation and Compatibility](https://www.nexusmods.com/skyrimspecialedition/articles/2761)

>  Installation:
> 
> - Load ICOW BEFORE OCOW. This is very important or the whole mod will be a mess.
> - DO NOT load OCW_CellSettings.esp. All cell settings are included in this patch.

I can't recall if any patches from OCW require `OCW_CellSettings.esp`, if they do then we'd need to disable those patch messages.

> - This mod should be towards the end of your load order with other compatibility patches as it is a patch at its core.

`UltimateCollege.esp` would need to have the same sorting rules as `OCW_CellSettings.esp`.

```yaml
  - name: 'OCW_CellSettings.esp'
    group: *cellSettingsGroup
    after:
      - 'ELE_SSE.esp'
      - 'ELFXEnhancer.esp'
      - 'ELFX - Hardcore.esp'
      - 'Immersive Music.esp'
      - 'Immersive Music Replacer.esp'
      - 'Luminosity Lighting Overhaul.esp'
      - 'Luminosity - Skyrim is Lit Edition.esp'
      - 'Soulmancer Soundtrack - NoVindDun.esp'
      - 'Unique Region Names.esp'
      # Group afters, to prevent sorting errors
      - 'MissingEZs_AllExteriors.esp'
      - 'MissingEZsFixed.esp'
      - 'NoGrassINCaves.esp'
````

File: masterlist.yaml [hunk 9]
```diff
@@ -18091,6 +18108,15 @@ plugins:
     after:
       - 'Qw_ACE_CRF Patch.esp'
       - 'Qw_WACCF_CRF Patch.esp'
+          
+  - name: 'UltimateCollege.esp'
+    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/44797' ]
```
Comment (MacSplody) [inline masterlist.yaml]: To match sorting rules for `OCW_CellSettings.esp`.
```suggestion
    url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/44797' ]
    group: *cellSettingsGroup
    after:
      - 'ELE_SSE.esp'
      - 'ELFXEnhancer.esp'
      - 'ELFX - Hardcore.esp'
      - 'Immersive Music.esp'
      - 'Immersive Music Replacer.esp'
      - 'Luminosity Lighting Overhaul.esp'
      - 'Luminosity - Skyrim is Lit Edition.esp'
      - 'Soulmancer Soundtrack - NoVindDun.esp'
      - 'Unique Region Names.esp'
      # Group afters, to prevent sorting errors
      - 'MissingEZs_AllExteriors.esp'
      - 'MissingEZsFixed.esp'
      - 'NoGrassINCaves.esp'
```

**Review by MacSplody — CHANGES_REQUESTED** [review]
I left a few suggestions for changes.
